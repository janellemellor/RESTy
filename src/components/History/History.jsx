import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';

const History = ({ history }) => {
  const allHistory = history.map((item, i) => (
    <li key={i}>
      <HistoryItem {...item} />
    </li>
  ));

  return (
    <ul>
      {allHistory}  
    </ul>
  );
};

History.propTypes = {
  history: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired,
    method: PropTypes.string.isRequired
  })).isRequired
};

export default History;
