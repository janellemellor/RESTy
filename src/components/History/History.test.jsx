import React from 'react';
import { shallow } from 'enzyme';
import History from './History';

describe('History Component', () => {
  it('matches a snapshot', () => {
    const requestHistory = [
      {
        url: 'https://github.com/janellemellor',
        method: 'GET'
      },
      {
        url: 'https://github.com/janellemellor',
        method: 'POST'
      }
    ]; 

    const wrapper = shallow(<History history={requestHistory} />);
    expect(wrapper).toMatchSnapshot();
  });  
});

