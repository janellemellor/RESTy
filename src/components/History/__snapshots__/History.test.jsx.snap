// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History Component matches a snapshot 1`] = `
<ul>
  <li
    key="0"
  >
    <HistoryItem
      method="GET"
      url="https://github.com/janellemellor"
    />
  </li>
  <li
    key="1"
  >
    <HistoryItem
      method="POST"
      url="https://github.com/janellemellor"
    />
  </li>
</ul>
`;
